indico/web/http_api/handlers.py

```diff
@@ -128,16 +128,20 @@ def handler(prefix, path):
     onlyPublic = get_query_parameter(queryParams, ['op', 'onlypublic'], 'no') == 'yes'
     onlyAuthed = get_query_parameter(queryParams, ['oa', 'onlyauthed'], 'no') == 'yes'
     scope = 'read:legacy_api' if request.method == 'GET' else 'write:legacy_api'
-    try:
-        oauth_valid, oauth_request = oauth.verify_request([scope])
-        if not oauth_valid and oauth_request and oauth_request.error_message != 'Bearer token not found.':
-            raise BadRequest('OAuth error: {}'.format(oauth_request.error_message))
-        elif g.get('received_oauth_token') and oauth_request.error_message == 'Bearer token not found.':
-            raise BadRequest('OAuth error: Invalid token')
-    except ValueError:
-        # XXX: Dirty hack to workaround a bug in flask-oauthlib that causes it
-        #      not to properly urlencode request query strings
-        #      Related issue (https://github.com/lepture/flask-oauthlib/issues/213)
+    authHeader = request.headers.get('Authorization')
+    if authHeader and authHeader[:5].lower() != 'basic':
```

Comment:
We don't use camelCase anymore. Those places where we do are legacy leftovers.

And I think it'd be nicer to combine those two lines into one:

```python
    if request.headers.get('Authorization', '').lower().startswith('basic ')
```

Reply:
Oh sorry about the camelCase. I mimicked the code just before the patch.

I can combine the two lines but it would then do the OAuth when there's no `Authorization` header present.

Also,
```
authHeader[:5].lower() != 'basic'
```
is faster than
```
request.headers.get('Authorization', '').lower().startswith('basic ')
```
see https://stackoverflow.com/questions/13578916/case-insensitive-string-startswith-in-python

HTTP Headers may be up to 16K on some servers.

What do you think?